Python/5_smallest_multiple.py

```diff
@@ -1,13 +1,11 @@
 # Problem 5
-def multiplies(n):
-	a = [i for i in xrange(11,21) if n%i==0]
+def get_multiplies(n, multiplies):
+	a = [i for i in xrange(1,multiplies+1) if n%i==0]
 	return a
 
-def smallest_multiple():
-	# primes product - 90 (number will ends with 00)
-	primes = 2*3*5*7*11*13*17*19-90
-	for i in xrange(primes,primes**2, 40):
-		if len(multiplies(i))==10:
-			print i
-			return
-smallest_multiple()
\ No newline at end of file
+def smallest_multiple(multiplies):
+	for i in xrange(2,multiplies**10, 2):
+		if len(get_multiplies(i, multiplies))==multiplies:
+			return i
+
+# print smallest_multiple(20)
```

Comment:
Try to solve this task in this way:

1 = 1
2 = 2
3 = 3
4 = 2*2
5 = 5
6 = 2*3
7 = 7
8 = 2*2*2
9 = 3*3
10 = 2*5

Primes need to remain
2: 3 times
3: 2 times
5: 1
7: 1
2^3*3^2*5*7 = 2520